db/migrations/20201020182259_backfill_kpack_lifecycle_data_association_to_droplets.rb

```diff
@@ -0,0 +1,18 @@
+Sequel.migration do
+  up do
+    unassociated_kpack_lifecycle_data = self[:kpack_lifecycle_data].select(:id, :build_guid).where(droplet_guid: nil).exclude(build_guid: nil)
+
+    unassociated_kpack_lifecycle_data.each do |kld|
+      droplet = self[:droplets].select(:guid).where(build_guid: kld[:build_guid]).first
+
+      # there may not have been a droplet created yet for the build
```

Comment:
I think there is an unlikely (?) race condition here where the `ccdb-migrate` job runs this but there are still `cf-api-servers` out there with the old code making "docker" droplets. If there are any apps in this state, they won't be able to get stack updates until the app is repushed.

I'm not sure if there is a way around that apart from either:
1. Reverting https://github.com/cloudfoundry/cloud_controller_ng/pull/1917/commits/91769482589608ed0ae664da9e8b33b70c4f29ff and always supporting the ability to update image for "docker" and "kpack" droplets.
2. Running the migration again as part of a future release.

Reply:
Resolving this, let's move discussion to this thread:
https://github.com/cloudfoundry/cloud_controller_ng/pull/1917#discussion_r508889764